reference/dic_tags.rst

```diff
@@ -227,6 +228,107 @@ assetic.templating.twig
 The tagged service will be removed from the container if
 ``framework.templating.engines`` config section does not contain ``twig``.
 
+auto_alias
+----------
+
+.. versionadded:: 2.7
+    The ``auto_alias`` tag was introduced in Symfony 2.7.
+
+**Purpose**: Define aliases based on the value of container parameters
+
+Consider the following configuration that defines three different but related
+services:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        services:
+            app.mysql_lock:
+                class: AppBundle\Lock\MysqlLock
+            app.postgresql_lock:
+                class: AppBundle\Lock\PostgresqlLock
+            app.sqlite_lock:
+                class: AppBundle\Lock\SqliteLock
+
+    .. code-block:: xml
+
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+            <services>
+                <service id="app.mysql_lock" class="AppBundle\Lock\MysqlLock" />
+                <service id="app.postgresql_lock" class="AppBundle\Lock\PostgresqlLock" />
+                <service id="app.sqlite_lock" class="AppBundle\Lock\SqliteLock" />
```

Comment:
these service should probably be private, so that the ones not aliased as ``app.lock`` will get removed when optimizing the container.
If you use auto-aliasing to choose the implementation, you should never access ``app.mysql_lock`` directly in your codebase anyway.